### ISSUE TITLE
Sections should not immediately reorganize their contents when items are removed

### ISSUE THREAD

**Issue by ntowle** [Cidan/BetterBags#59]
Consider this usecase:

I have some items in a section, and I want to sell the first 6 of them to a vendor:
![image](https://github.com/Cidan/BetterBags/assets/20693723/9c9bfff1-6d6d-4e03-b43e-ae40451f6526)

The fastest way for me to do this is to right-click each of the first 6 items, moving my mouse to each item. But after clicking the first couple of them the section will reflow, and now I've lost my place and am vendoring the wrong things. I think it would be better if item positions remained static when transferring items out of your bags (vendoring, moving to bank, equipping, etc).

**Comment by FlyingDiver** [comment]
That's bit me too.  I think re-flowing as items are added is OK, since you have to put them somewhre, but I don't think it's necessary when items are removed.

**Comment by SpareSimian** [comment]
Instead of reflowing on remove, just enable the "New" button so we can force a reflow when we're done. (Or it can reflow when the bag is closed and reopened.)

**Comment by Cidan** [comment]
This is solid feedback. I'll evaluate how to stop items from re-sorting if they are removed, such that they will just leave gaps, and resort on next bag open.

**Comment by deathmessinger** [comment]
It doesn't just do it when selling, does it when you equipping too 
(caused my pally to use cloth hands for about an hour)

**Comment by Cidan** [comment]
I just pushed a change for this in grid mode, where items won't reshuffle if you move items out of your bag. They will reshuffle in a few cases though:

1) If you completely empty out a section
2) If you add any item to your bag
3) If you close and reopen the bag

I think this is a solid experience. Note, this only happens in grid mode. I likely won't implement this in list mode, as it doesn't make much sense. I'll probably implement this in one bag view mode after I make sure there are no bugs.

Please test it and let me know what you think! Thanks!

**Comment by deathmessinger** [comment]
ok this seems to stop rearranging inside the groups but the groups move all over after selling one thing from the group

**Comment by Cidan** [comment]
> ok this seems to stop rearranging inside the groups but the groups move all over after selling one thing from the group

Yes, I noticed that too last night -- I'll be working on that as well.

**Comment by Cidan** [comment]
This has been fixed and released for both bank and bags. Both items, and sections, will not reorder when removing items from the bags.

Thanks again for the feedback!

**Comment by ItsDandyDigital** [comment]
This reordering happens each time you withdraw something from your bank. The bag and bank gets reshuffled. Any way you can simply add a toggle to disable this experience, or make it only reorg when opening the bank/bags, or clicking the reorganize button?

**Comment by SpareSimian** [comment]
The bags will need to be reorganized if there's insufficient space when an item moves. Was there enough space?

**Comment by ItsDandyDigital** [comment]
> The bags will need to be reorganized if there's insufficient space when an item moves. Was there enough space?

Both reagent bank, and bags have space. Moving an item from bags to reagent bank -- no reorg. Moving an item from reagent bank to the bags -- reorg with each item moved.

**Comment by Cidan** [comment]
I'm actively working on this for the bank. I have a branch going -- just need to fix a few last bugs. Sit tight.

**Comment by ergh99** [comment]
Specific experience example from Remix:
* I click a stack of prismatic gems, wanting to do this 3 times in a row to upgrade 2 steps
* after the first click, there's a new gem item in the category which has now jumped to a completely different spot on my screen
* I now have to hunt down the category and find the correct gem stack again
This is very annoying.

**Comment by galaarelune** [comment]
As of 9/6/24 with latest version, this is still happening for me.  I was moving things from my bags to the reagent bank and the warband bank and the section kept jumping from top to bottom with each item.  I move an item in, it moves to the top.  I move another item in, it goes back to where it was.  Repeated over and over.  Makes it really annoying to move things.

**Comment by Cidan** [comment]
Fixed in the latest alpha.
